### ISSUE TITLE
resolveError not working for "required" and "string" validators

### ISSUE THREAD

**Issue by laurentpayot** [jagi/meteor-astronomy#366]
Hi, I love the new V2. I'm using `resolveError` to generate translated error messages in forms.

I noticed that `resolveError` is not taken into account at least for `required` and `string` validators, and maybe for all validators without parameters because `minLength` and other validators with parameters are working.

For instance (in _CoffeeScript_):

``` coffeescript
TasksCol = new Mongo.Collection('tasks')
Task = Astro.Class.create(
    name: 'Task'
    collection: TasksCol
    fields:
        title:
            type: String
            validators: [
                {type: 'required', resolveError: (args) -> "required!"}
                {type: 'string', resolveError: (args) -> "string!"}
                {type: 'minLength', param: 10, resolveError: (args) -> "minLength!"}
            ]
)
```

In _meteor shell_ for the `required` validator we get the default error message instead of "required!":

```
> t = new Astro.Class.classes.Task(); t.validate({fields: ['title']})
Error: "title" is required [validation-error]
    at new _class (packages/mdg:validation-error/validation-error.js:14:12)
    at throwError (packages/jagi:astronomy/lib/modules/validators/validator.js:62:15)
    at Validator.validate (packages/jagi:astronomy/lib/modules/validators/validator.js:88:9)
    at Object._validators2.default.(anonymous function) [as required] (packages/jagi:astronomy/lib/modules/validators/validator.js:100:17)
    at ScalarField.validate (packages/jagi:astronomy/lib/modules/fields/field.js:31:18)
    at ScalarField.validate (packages/jagi:astronomy/lib/modules/fields/scalar_field.js:12:5)
    at packages/jagi:astronomy/lib/modules/validators/utils/document_validate.js:109:13
    at catchValidationError (packages/jagi:astronomy/lib/modules/validators/utils/document_validate.js:38:7)
    at packages/jagi:astronomy/lib/modules/validators/utils/document_validate.js:67:28
    at arrayEach (/home/laurent/.meteor/packages/jagi_astronomy/.2.0.0.1nuuhzl++os+web.browser+web.cordova/npm/node_modules/lodash/lodash.js:498:11)
```

Same for the `string` validator:

```
t = new Astro.Class.classes.Task({title: 1}); t.validate({fields: ['title']})
Error: "title" has to be a string [validation-error]
    at new _class (packages/mdg:validation-error/validation-error.js:14:12)
    at throwError (packages/jagi:astronomy/lib/modules/validators/validator.js:62:15)
    at Validator.validate (packages/jagi:astronomy/lib/modules/validators/validator.js:88:9)
    at Object._validators2.default.(anonymous function) [as string] (packages/jagi:astronomy/lib/modules/validators/validator.js:100:17)
    at Type.validate (packages/jagi:astronomy/lib/modules/fields/types/string.js:8:16)
    at ScalarField.validate (packages/jagi:astronomy/lib/modules/fields/scalar_field.js:18:24)
    at packages/jagi:astronomy/lib/modules/validators/utils/document_validate.js:109:13
    at catchValidationError (packages/jagi:astronomy/lib/modules/validators/utils/document_validate.js:38:7)
    at packages/jagi:astronomy/lib/modules/validators/utils/document_validate.js:67:28
    at arrayEach (/home/laurent/.meteor/packages/jagi_astronomy/.2.0.0.1nuuhzl++os+web.browser+web.cordova/npm/node_modules/lodash/lodash.js:498:11)
```

but `resolveError` is working for the `minLength` validator:

```
> t = new Astro.Class.classes.Task({title: "foo"}); t.validate({fields: ['title']})
Error: minLength! [validation-error]
    at new _class (packages/mdg:validation-error/validation-error.js:14:12)
    at throwError (packages/jagi:astronomy/lib/modules/validators/validator.js:62:15)
    at Validator.validate (packages/jagi:astronomy/lib/modules/validators/validator.js:74:9)
    at _validators2.default.(anonymous function) (packages/jagi:astronomy/lib/modules/validators/validator.js:100:17)
    at packages/jagi:astronomy/lib/modules/validators/utils/document_validate.js:123:13
    at arrayEach (/home/laurent/.meteor/packages/jagi_astronomy/.2.0.0.1nuuhzl++os+web.browser+web.cordova/npm/node_modules/lodash/lodash.js:498:11)
    at Function.forEach (/home/laurent/.meteor/packages/jagi_astronomy/.2.0.0.1nuuhzl++os+web.browser+web.cordova/npm/node_modules/lodash/lodash.js:8382:11)
    at packages/jagi:astronomy/lib/modules/validators/utils/document_validate.js:117:9
    at catchValidationError (packages/jagi:astronomy/lib/modules/validators/utils/document_validate.js:38:7)
    at packages/jagi:astronomy/lib/modules/validators/utils/document_validate.js:67:28
```

**Comment by lukejagodzinski** [comment]
`string` and `required` validators are added automatically to your field so adding it one more time will have no effect. The only way to generate custom error message in that way is using the `resolveError` method on the level of the class, what I forgot to describe in the documentation. I will update it.

``` js
const Task = Class.create({
  name: 'Task',
  collection: Tasks,
  /* ... */
  resolveError(args) {
  }
});
```

Examine what arguments are passed to this method and you will see how to generate error message. I will update the docs.

**Comment by laurentpayot** [comment]
I examined the arguments passed to this class function with one invalid field with a `string` validator and with another invalid field with a `number` validator. The arguments passed were the same except for the `name`, `nestedName` and `value`. How can I know that the first invalid value had to be a string? Is it possible to add a `validatorName` or `type` argument?

**Comment by lukejagodzinski** [comment]
Ahhh I forgot to pass validator name to the `resolveError` method. I've fixed it and I will publish 2.0.1 version. It will be available under the `validator` property

**Comment by laurentpayot** [comment]
Perfect. Thanks Jagi!
